File: CycleChesscpp/app/src/main/java/com/serebryakov/cyclechesscpp/application/model/game/MoveType.kt
package com.serebryakov.cyclechesscpp.application.model.game

enum class MoveType {
    CASTLING, // рокировка
    PASSANT, // взятие на проходе
    MAGIC_PAWN_TRANSFORMATION, // превращение пешки
    NOT_SPECIAL, // обычный ход
    NOT_MOVE, // ошибка
}

fun String.ToMoveType(): MoveType {
    return when(this.toInt()) {
        0 -> MoveType.CASTLING
        1 -> MoveType.PASSANT
        2 -> MoveType.MAGIC_PAWN_TRANSFORMATION
        3 -> MoveType.NOT_SPECIAL
        4 -> MoveType.NOT_MOVE
        else -> {
            MoveType.NOT_MOVE
        }
    }
}

File: CycleChessBot/game/move_type.py
from enum import Enum

class MoveType(Enum):
    CASTLING = 0
    PASSANT = 1
    MAGIC_PAWN_TRANSFORMATION = 2
    NOT_SPECIAL = 3
    NOT_MOVE = 4

    def __str__(self) -> str:
        return str(self.value)
